spec/requests/client_visibility_spec.rb

```diff
@@ -5,839 +5,896 @@
 RSpec.describe ClientsController, type: :request do
   include_context 'visibility test context'
 
-  context 'when config b is in affect' do
-    before do
-      GrdaWarehouse::Config.delete_all
-      GrdaWarehouse::Config.invalidate_cache
-    end
-    let!(:config) { create :config_b }
-    let!(:user) { create :user }
+  configs_variations = GrdaWarehouse::Config.available_cas_methods.values.product( # cas available method
+    [true, false], # project type override
+    GrdaWarehouse::Config.family_calculation_methods.values, # family calculation method
+    GrdaWarehouse::Config.available_release_durations, # release duration
+  )
```

Comment:
These should encompass only options that are not covered in `spec/factories/grda_warehouse/config.rb` since those are known configurations that the tests below confirm are working as expected.
The idea here is good though.  For now let's test:
```
cas_available_method
consent_visible_to_all
expose_coc_code
health_emergency
health_emergency_tracing
health_priority_age
multi_coc_installation
```

Reply:
If you can figure out how to make the product return a bunch of hashes, with all possible options, that will make choosing names for things much more clear.